src/commandcollection.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module core/commandcollection
+ */
+
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+
+/**
+ * Collection of commands. Its instance is available in {@module:core/editor/editor~Editor#commands}.
+ */
+export default class CommandCollection {
+	/**
+	 * Creates collection instance.
+	 */
+	constructor() {
+		/**
+		 * Command map.
+		 *
+		 * @private
+		 * @member {Map}
+		 */
+		this._commands = new Map();
+	}
+
+	/**
+	 * Registers a new command.
+	 *
+	 * @param {String} commandName The name of the command.
+	 * @param {module:core/command~Command} command
+	 */
+	add( commandName, command ) {
+		this._commands.set( commandName, command );
+	}
+
+	/**
+	 * Retrieves a command from the collection.
+	 *
+	 * @param {String} commandName The name of the command.
+	 * @returns {module:core/command~Command}
+	 */
+	get( commandName ) {
+		return this._commands.get( commandName );
+	}
+
+	/**
+	 * Executes a command.
+	 *
+	 * @param {String} commandName The name of the command.
+	 */
+	execute( commandName, ...args ) {
```

Comment:
I mean, I like `editor.commands.get( 'bold' ).execute( { forceState: true } );` because it's clear here that `{ forceState: true }` is just a parameter of the `execute` method. Nothing more, nothing less.